### ISSUE TITLE
Make Console rendering deterministic and terminal-safe

### ISSUE THREAD

**Issue by Extender92** [Extender92/Monopoly#20]
## Problem/Goal

Console rendering does not retain enough layout information to erase or redraw prior content reliably. Variable-sized menus, cards, logs, and terminal constraints can leave stale text or overwrite adjacent regions.

## Required behavior

- All terminal reads, writes, sizing, cursor movement, and clearing go through `IConsoleWrapper`.
- Renderers track each region's origin, width, and height and clear the union needed before redraw.
- A card footer starts below the last body row and never overwrites it.
- Replacing a wider or taller menu, card, or log completely erases the previous content.
- Unsupported or undersized terminals produce a clear frontend error instead of corrupt output or an unhandled exception.

## Scope

- Console rendering/layout bookkeeping and terminal capability checks.
- Fake-wrapper tests for redraw, clear regions, footer placement, and error behavior.
- Presentation-only Console changes.

## Out of scope

- Moving game rules or authoritative state into Console.
- A new frontend, terminal library replacement, or visual redesign.
- Core event/notification architecture, owned by #50.

## Acceptance criteria

- No production renderer bypasses `IConsoleWrapper` for terminal access.
- Region bookkeeping correctly handles shorter, taller, narrower, and wider replacement content.
- Card footer tests prove there is no last-row off-by-one overwrite.
- Previous menu, card, and log content is fully cleared before replacement.
- Unsupported cursor/size behavior and insufficient dimensions return an actionable Console error.
- Fake-wrapper tests assert exact positions and clear extents without a real terminal.
- Console contains no new rule decisions or Core mutations.
- Full Release build and test suite pass without new warnings.

## Dependencies

- None; may proceed in parallel with #16–#18.

## Related issues

- Related to #31
- Related to #50

**Comment by Extender92** [comment]
Deferred during neutral-Core backlog sanitation. This issue has not been implemented. It will not be copied to the clean repository and is not a cutover prerequisite. Reconsider the behavior only as a new, independently scoped neutral issue after cutover, based on the clean architecture and current product goals.
